DotNet Libraries.txt

```diff
@@ -0,0 +1,272 @@
+OBJECT Codeunit 50500 DateTime Formatting  Library
+{
+  OBJECT-PROPERTIES
+  {
+    Date=06.09.17;
+    Time=16:33:04;
+    Modified=Yes;
+    Version List=;
+  }
+  PROPERTIES
+  {
+    OnRun=BEGIN
+          END;
+
+  }
+  CODE
+  {
+
+    PROCEDURE GetStandardDateTimeString@1(CurrentDateTime@1001 : DateTime;Pattern@1000 : Text) : Text;
+    VAR
+      DateTime@1100408000 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.DateTime";
+      CultureInfo@1100408001 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.Globalization.CultureInfo";
+    BEGIN
+      DateTime := CurrentDateTime;
+      EXIT(DateTime.ToString(Pattern, CultureInfo.InvariantCulture));
+    END;
+
+    PROCEDURE EvaluateStandardDateTimeString@2(DateTimeString@1100408000 : Text;Pattern@1000 : Text) : DateTime;
+    VAR
+      DateTimeOffset@1100408002 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.DateTimeOffset";
+      CultureInfo@1100408001 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.Globalization.CultureInfo";
+      Convert@1100408004 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.Convert";
+    BEGIN
+      DateTimeOffset := DateTimeOffset.ParseExact(DateTimeString,Pattern,CultureInfo.InvariantCulture);
+      DateTimeOffset := DateTimeOffset.ToLocalTime;
+      EXIT(DateTimeOffset.DateTime);
+    END;
+
+    PROCEDURE GetUTCDateTimeString@3() : Text;
+    VAR
+      DateTime@1000 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.DateTime";
+    BEGIN
+      EXIT(GetStandardDateTimeString(GetUTCDateTime,'R'));
+    END;
+
+    PROCEDURE EvaluateUTCDateTimeString@4(DateTimeString@1100408000 : Text) : DateTime;
+    BEGIN
+      EXIT(EvaluateStandardDateTimeString(DateTimeString,'R'));
+    END;
+
+    PROCEDURE GetUTCDateTime@5() : DateTime;
+    VAR
+      DateTime@1000 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.DateTime";
+    BEGIN
+      DateTime := DateTime.Now;
+      EXIT(DateTime)

```

Comment:
Allocation is unnecessary

EXIT(DateTime.Now);

Reply:
Should be EXIT(DateTime.UTCNow);